packages/medusa/src/services/price-list.ts

```diff
@@ -336,6 +345,28 @@ class PriceListService extends BaseService {
       return [productsWithPrices, count]
     })
   }
+
+  /**
+   * Add `currency_code` to an MA record if `region_id`is passed.
+   * @param prices - a list of PriceListPrice(Create/Update)Input records
+   * @return {Promise} updated `prices` list
+   */
+  private async addCurrencyFromRegion<
+    T extends PriceListPriceUpdateInput | PriceListPriceCreateInput
+  >(prices: T[]): Promise<T[]> {
+    const prices_: typeof prices = []
+
+    for (const p of prices) {
+      if (p.region_id) {
+        const region = await this.regionService_.retrieve(p.region_id)
```

Comment:
since it is a protected/private method there is indeed no need for the atomic phase.
But, here we should call the regionService like this (to be in the transaction):

```ts
this.regionService_.withTransaction(this.manager_).retrieve(price.region_id)
```

Reply:
is there any reason that we do not mutate the prices from the input parameter and return themselves instead?

Reply:
No particular reason other than just following immutable practices.